astronomy/orbit_analysis_test.cpp

```diff
@@ -0,0 +1,509 @@
+﻿
+#include <string>
+#include <vector>
+#include <utility>
+
+#include "astronomy/orbit_recurrence.hpp"
+#include "astronomy/orbital_elements.hpp"
+#include "astronomy/standard_product_3.hpp"
+#include "base/not_null.hpp"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "physics/body_centred_non_rotating_dynamic_frame.hpp"
+#include "physics/ephemeris.hpp"
+#include "physics/solar_system.hpp"
+#include "testing_utilities/matchers.hpp"
+#include "testing_utilities/is_near.hpp"
+
+namespace principia {
+namespace astronomy {
+
+using base::make_not_null_unique;
+using base::not_null;
+using geometry::Instant;
+using geometry::Position;
+using integrators::SymmetricLinearMultistepIntegrator;
+using integrators::methods::QuinlanTremaine1990Order12;
+using physics::BodyCentredNonRotatingDynamicFrame;
+using physics::BodySurfaceDynamicFrame;
+using physics::DiscreteTrajectory;
+using physics::Ephemeris;
+using physics::MasslessBody;
+using physics::RotatingBody;
+using physics::SolarSystem;
+using quantities::Mod;
+using quantities::Time;
+using quantities::astronomy::JulianYear;
+using quantities::si::ArcMinute;
+using quantities::si::ArcSecond;
+using quantities::si::Day;
+using quantities::si::Degree;
+using quantities::si::Kilo;
+using quantities::si::Metre;
+using quantities::si::Micro;
+using quantities::si::Milli;
+using quantities::si::Minute;
+using quantities::si::Radian;
+using quantities::si::Second;
+using testing_utilities::IsNear;
+using testing_utilities::IsOk;
+using ::testing::AllOf;
+using ::testing::Property;
+
+namespace {
+
+struct SP3Files {
+  std::vector<std::string> names;
+  StandardProduct3::Dialect dialect;
+
+  static SP3Files const& GNSS() {
+    static const SP3Files files = {{"WUM0MGXFIN_20190970000_01D_15M_ORB.SP3",
+                                    "WUM0MGXFIN_20190980000_01D_15M_ORB.SP3",
+                                    "WUM0MGXFIN_20190990000_01D_15M_ORB.SP3",
+                                    "WUM0MGXFIN_20191000000_01D_15M_ORB.SP3",
+                                    "WUM0MGXFIN_20191010000_01D_15M_ORB.SP3",
+                                    "WUM0MGXFIN_20191020000_01D_15M_ORB.SP3",
+                                    "WUM0MGXFIN_20191030000_01D_15M_ORB.SP3",
+                                    "WUM0MGXFIN_20191040000_01D_15M_ORB.SP3",
+                                    "WUM0MGXFIN_20191050000_01D_15M_ORB.SP3",
+                                    "WUM0MGXFIN_20191060000_01D_15M_ORB.SP3"},
+                                   StandardProduct3::Dialect::ChineseMGEX};
+    return files;
+  }
```

Comment:
Please put the definitions of these function offline at the end of the unnamed namespace, i.e. keep the struct declaration clean.

Reply:
Done.